### PR TITLE
Add Debian packaging support.

### PR DESCRIPTION
**Pull request by cathay4t**
* `make deb` would do the magic.
 * Due to [debian bug](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=657401), the output folder are forced to `../`.
 * No multi-arch support yet, just found Debian multi-arch wiki too long to read.
 * No python3 support due to missing of `python3-pywbem` in Debian repo.
 * Only support debian unstable and testing. (debian stable 9.3 don't ship `libsqlite3-dev`)

### CODE REVIEW

**Comment by cathay4t** [comment]
@rickysarraf @joehandzik Could you take a look on this? Thank you.

**Comment by cathay4t** [comment]
V2 uploaded:
 * Add missing `tmpfiles.d/lsm-tmpfiles.conf` and `/etc/lsm/pluginconf.d/`.

**Comment by cathay4t** [comment]
The ppc63 CI server is down.

The failure of ontap CI is caused by expired license.

Besides above, all CI passed.

**Comment by cathay4t** [comment]
@seitzbg Could you try this PR to see whether it fit your expectation? Thank you.

**Comment by tasleson** [comment]
@cathay4t Have we followed up with lab folks to see if we can get some new licenses for ontap?

**Comment by cathay4t** [comment]
@tasleson Request sent. Meanwhile, let me file out whether I can run a ONTAP simulator in libvirt.

**Comment by cathay4t** [comment]
ppc CI server up, retry CI.

File: debian/gbp.conf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[DEFAULT]
```
Comment (rickysarraf) [inline debian/gbp.conf]: You can delete this file, debian/gbp.conf. This is custom configuration for my use case and has nothing to do with the package build.

Comment (cathay4t) [reply debian/gbp.conf]: Sure. Thanks.

**Comment by seitzbg** [comment]
Thanks for this work!  Any ETA for merge?

**Comment by cathay4t** [comment]
I will test it again next week. If there is no problem found, I will merge it.

**Comment by seitzbg** [comment]
Thanks!

**Comment by cathay4t** [comment]
Tested on latest debian testing(sid). Merged.

**Comment by seitzbg** [comment]
Having this issue when trying to build on Ubuntu 16.04 (Xenial):

```make[1]: Leaving directory '/home/seitz/git/libstoragemgmt'
        rm -f debian/*.debhelper.log
   dh_auto_build
        make -j1
make[1]: Entering directory '/home/seitz/git/libstoragemgmt'
make  all-recursive
make[2]: Entering directory '/home/seitz/git/libstoragemgmt'
Making all in c_binding
make[3]: Entering directory '/home/seitz/git/libstoragemgmt/c_binding'
Making all in include
make[4]: Entering directory '/home/seitz/git/libstoragemgmt/c_binding/include'
Making all in libstoragemgmt
make[5]: Entering directory '/home/seitz/git/libstoragemgmt/c_binding/include/libstoragemgmt'
make[5]: Nothing to be done for 'all'.
make[5]: Leaving directory '/home/seitz/git/libstoragemgmt/c_binding/include/libstoragemgmt'
make[5]: Entering directory '/home/seitz/git/libstoragemgmt/c_binding/include'
make[5]: Nothing to be done for 'all-am'.
make[5]: Leaving directory '/home/seitz/git/libstoragemgmt/c_binding/include'
make[4]: Leaving directory '/home/seitz/git/libstoragemgmt/c_binding/include'
make[4]: Entering directory '/home/seitz/git/libstoragemgmt/c_binding'
make[4]: Nothing to be done for 'all-am'.
make[4]: Leaving directory '/home/seitz/git/libstoragemgmt/c_binding'
make[3]: Leaving directory '/home/seitz/git/libstoragemgmt/c_binding'
Making all in python_binding
make[3]: Entering directory '/home/seitz/git/libstoragemgmt/python_binding'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/seitz/git/libstoragemgmt/python_binding'
Making all in plugin
make[3]: Entering directory '/home/seitz/git/libstoragemgmt/plugin'
Making all in simc
make[4]: Entering directory '/home/seitz/git/libstoragemgmt/plugin/simc'
  CCLD     simc_lsmplugin
/usr/bin/ld: vector.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
vector.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:456: recipe for target 'simc_lsmplugin' failed
make[4]: *** [simc_lsmplugin] Error 1
make[4]: Leaving directory '/home/seitz/git/libstoragemgmt/plugin/simc'
Makefile:816: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/home/seitz/git/libstoragemgmt/plugin'
Makefile:723: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/home/seitz/git/libstoragemgmt'
Makefile:617: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/seitz/git/libstoragemgmt'
dh_auto_build: make -j1 returned exit code 2
debian/rules:13: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1376:
dpkg-buildpackage -rfakeroot -D -us -uc -i -b failed
```

Any advice welcome!

**Comment by cathay4t** [comment]
@seitzbg I failed to reproduce your problem. But I found other problem `debuilder < 11`.

I have merged two fixes to current master and tested in docker for Ubuntu 16.04 LTS and Ubuntu 18.04 LTS.

Try these commands:

```bash
apt-get install `cat deb_dependency`
./autogen.sh
./configure
make deb
```

**Comment by cathay4t** [comment]
@seitzbg Did I solve your problem?

**Comment by seitzbg** [comment]
Yes make deb worked fine.

On Sun, May 13, 2018 at 11:17:42AM -0700, Gris Ge wrote:
> @seitzbg Did I solve your problem?

-- 
             
Bryan G. Seitz

**Pull request merged**